sansio_lsp_client/client.py

```diff
@@ -171,31 +360,63 @@ def parse_request(event_cls: t.Type[Event]) -> Event:
                     " or ServerNotification"
                 )
 
-        if request.method == "window/showMessage":
+        if request.method == "workspace/workspaceFolders":
+            return parse_request(WorkspaceFolders)
+
+        elif request.method == "workspace/configuration":
+            return parse_request(ConfigurationRequest)
+
+        elif request.method == "window/showMessage":
             return parse_request(ShowMessage)
         elif request.method == "window/showMessageRequest":
             return parse_request(ShowMessageRequest)
         elif request.method == "window/logMessage":
             return parse_request(LogMessage)
+
         elif request.method == "textDocument/publishDiagnostics":
             return parse_request(PublishDiagnostics)
-        else:
-            raise NotImplementedError(request)
```

Comment:
Why was this deleted?

Reply:
It was moved out of the `if-elif` chain - to be raised when `WorkDoneProgress` has unrecognized `kind`

Reply:
Ignoring all unknown requests is bad. Please add it back somehow.